gremlin-server/src/test/java/org/apache/tinkerpop/gremlin/server/ClientWithOptionsTest.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.tinkerpop.gremlin.server;
+
+import org.apache.tinkerpop.gremlin.driver.Client;
+import org.apache.tinkerpop.gremlin.driver.Cluster;
+import org.apache.tinkerpop.gremlin.driver.Result;
+import org.apache.tinkerpop.gremlin.driver.remote.DriverRemoteConnection;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversal;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversalSource;
+import org.apache.tinkerpop.gremlin.structure.Vertex;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.ExecutionException;
+
+import static org.apache.tinkerpop.gremlin.process.traversal.AnonymousTraversalSource.traversal;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.fail;
+
+/**
+ * Tests for options sent by the client using with().
+ *
+ * @author Ryan Tan
+ */
+public class ClientWithOptionsTest extends AbstractGremlinServerIntegrationTest {
+
+    @Test
+    public void shouldTimeOutAliasedClientSendingBytecode() {
+        final Cluster cluster = TestClientFactory.build().create();
+        Client client = cluster.connect().alias("ggrateful");
+        GraphTraversalSource g = traversal().withRemote(DriverRemoteConnection.using(client, "g"));
+        GraphTraversal traversal = g.with("evaluationTimeout", 1).V().both().both().both();
```

Comment:
Approximately how long does this request take to complete? Are we going to add one  more flaky test?

Reply:
From testing before and after each query with `System.currentTimeMillis()`, the bytecode queries return after 300ms+ and the script queries return after 10000ms+ consistently.  Since this is well over the timeout of 1ms, these tests will likely not flake.